x-pack/plugin/identity-provider/docs/en/rest-api/idp-saml-register-sp.asciidoc

```diff
@@ -0,0 +1,101 @@
+[role="xpack"]
+[[idp-saml-register-sp]]
+=== Register or update a SAML service provider
+++++
+<titleabbrev>Register or update a SAML service provider</titleabbrev>
+++++
+Registers a SAML service provider for use with this Identity Provider or updates the configuration of an existing one.
+
+[[idp-saml-register-sp-request]]
+==== {api-request-title}
+
+`POST /_idp/saml/sp/{sp_entity_id} +
+ PUT  /_idp/saml/sp/{sp_entity_id}`
+
+[[idp-saml-register-sp-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have a role that grants the `cluster:admin/idp/saml/sp` privilege.
+
+[[idp-saml-register-desc]]
+==== {api-description-title}
+
+This API registers a Service Provider with the Identity Provider and sets the necessary configuration options for that SP.
+
+
+[[idp-saml-register-sp-path-params]]
+==== {api-path-parms-title}
+
+`sp_entity_id::
+(string) The SAML entity Id of the service provider to be registered or updated. In case the entity Id is a URL, it should be urlencoded.
+
+[[idp-saml-register-sp-params]]
+==== {api-query-parms-title}
+
+`refresh`::
+(Optional, string) One of `true`, `false`, or `wait_for`.
+These values have the same meaning as in the <<docs-refresh, Index API>>,
+but the default value for this API is `false`.
+
+[[idp-saml-register-sp-body]]
+==== {api-request-body-title}
+
+The following parameters can be specified in the body of a POST or PUT request:
+
+`name`::
+(Required, string) A name to identify this service provider. Used only for informational purposes
+
+`entity_id`::
+(Required, string) The SAML entity Id of the service provider.
```

Comment:
This isn't required.
If it is not set, it will be populated from the URL parameter.
If it is set, it _must_ match the URL parameter.
https://github.com/elastic/elasticsearch/blob/8552c5e58a1ccfa9c451a70d791e7dc1b08e2851/x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/action/PutSamlServiceProviderRequest.java#L36-L43